app/Http/Controllers/StudentsCouncil/AnonymousQuestionController.php

```diff
@@ -0,0 +1,171 @@
+use Carbon\Carbon;
+use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Support\Facades\DB;
+use Maatwebsite\Excel\Facades\Excel;
+
+use App\Models\AnonymousQuestions\AnswerSheet;
+use App\Models\Semester;
+use App\Models\Question;
+use App\Models\QuestionOption;
+use App\Exports\UsersSheets\AnonymousQuestionsExport;
+
+/**
+ * Controls actions related to anonymous questions.
+ */
+class AnonymousQuestionController extends Controller
+{
+    /**
+     * Lists semesters as collapsible cards;
+     * containing the export option,
+     * the list of questions
+     * and the option to add new ones.
+     */
+    public function indexSemesters()
+    {
+        $this->authorize('administer', AnswerSheet::class);
+
+        return view('student-council.anonymous-questions.index_semesters');
+    }
+
+    /**
+     * Returns the 'new question' page.
+     */
+    public function create(Semester $semester)
+    {
+        $this->authorize('administer', AnswerSheet::class);
+
+        if ($semester->isClosed()) {
+            abort(403, "tried to add a question to a closed semester");
+        }
+        return view('student-council.anonymous-questions.create', [
+            "semester" => $semester
+        ]);
+    }
+
+    /**
+     * Saves a new question.
+     */
+    public function store(Request $request, Semester $semester)
+    {
+        $this->authorize('administer', AnswerSheet::class);
+
+        if ($semester->isClosed()) {
+            abort(403, "tried to add a question to a closed semester");
+        }
+
+        $validator = Validator::make($request->all(), [
+            'title' => 'required|string',
+            'has_long_answers' => 'nullable|in:on',
+            'max_options' => 'exclude_if:has_long_answers,on|required|min:1',
+            'options' => 'exclude_if:has_long_answers,on|required|array|min:1',
+            'options.*' => 'exclude_if:has_long_answers,on|nullable|string|max:255',
+        ]);
+        $hasLongAnswers = isset($request->has_long_answers);
+        if (!$hasLongAnswers) {
+            $options = array_filter($request->options, function ($s) {
+                return $s != null;
+            });
+            if (count($options) == 0) {
+                $validator->after(function ($validator) {
+                    $validator->errors()->add('options', __('voting.at_least_one_option'));
+                });
+            }
+        }
+        $validator->validate();
+
+        // NOTE: for now, we set the start and end dates for the _current_ semester;
+        // they will be updated when we set the dates for the semester evaluation form
+        $question = $semester->questions()->create([
+            'title' => $request->title,
+            'max_options' => $hasLongAnswers ? 0 : $request->max_options,
+            'has_long_answers' => $hasLongAnswers,
+            'opened_at' => (app(SemesterEvaluationController::class))->getStartDate() ?? Carbon::now(),
```

Comment:
didn't the solution I suggested to add HasPeriodicEvents here work?

Reply:
It does not really make sense. We have one periodicEvent which might belong to the semester you are looking for or not. The attached semester to the periodicEvent is only there to fix the semester used by the context. You should set the semester id of the feedback based on that, if you don't have that yet.